File: devops-scm-provider/devops-scm-provider-svn/devops-scm-provider-tsvn/src/main/java/com/tencent/devops/scm/provider/svn/tsvn/TSvnScmProvider.java
package com.tencent.devops.scm.provider.svn.tsvn;

import com.tencent.devops.scm.api.RepositoryService;
import com.tencent.devops.scm.api.WebhookParser;
import com.tencent.devops.scm.provider.svn.common.AbstractSvnScmProvider;

public class TSvnScmProvider extends AbstractSvnScmProvider {

    @Override
    public RepositoryService repositories() {
        return null;
    }

    @Override
    public WebhookParser webhookParser() {
        return null;
    }
}

File: devops-scm-provider/devops-scm-provider-svn/devops-scm-provider-tsvn/src/main/kotlin/com/tencent/devops/scm/provider/svn/tsvn/TSvnScmProvider.kt
package com.tencent.devops.scm.provider.svn.tsvn

import com.tencent.devops.scm.api.RefService
import com.tencent.devops.scm.api.RepositoryService
import com.tencent.devops.scm.api.WebhookParser
import com.tencent.devops.scm.provider.svn.common.AbstractSvnScmProvider
import com.tencent.devops.scm.sdk.common.connector.ScmConnector
import com.tencent.devops.scm.sdk.tsvn.TSvnApiFactory

class TSvnScmProvider : AbstractSvnScmProvider {
    private val apiFactory: TSvnApiFactory

    constructor(apiUrl: String, connector: ScmConnector) : this(TSvnApiFactory(apiUrl, connector))

    constructor(apiFactory: TSvnApiFactory) {
        this.apiFactory = apiFactory
    }

    override fun repositories(): RepositoryService = TSvnRepositoryService(apiFactory)

    override fun refs(): RefService = TSvnRefService(apiFactory)

    override fun webhookParser(): WebhookParser = TSvnWebhookParser()
}